server/src/main/java/org/opensearch/index/translog/RemoteFsTranslog.java

```diff
@@ -505,11 +543,7 @@ public static void cleanup(Repository repository, ShardId shardId, ThreadPool th
     }
 
     protected void onDelete() {
-        if (primaryModeSupplier.getAsBoolean() == false) {
```

Comment:
The delete is invoked from IndexShard where the primaryMode check is present already.

https://github.com/opensearch-project/OpenSearch/blob/5bb6caec906f9e89d330332ebb74789571409eb1/server/src/main/java/org/opensearch/index/shard/IndexShard.java#L1989-L1992

This code is only triggered in case of index deletion in which case we should let the deletion complete.

Here, we have added to assert that this can only be called by cluster applier thread. Will add a comment as well.